### PR TITLE
BOLT update, including payment metadata support

### PR DESCRIPTION
**Pull request by rustyrussell**
Thanks to Joost, bolt11 spec now supports metadata (see https://github.com/lightning/bolts/pull/912).  To get there, I updated the spec (in stages, there were other changes).

We don't issue invoices with metadata, but we *do* pay them properly if asked.

### CODE REVIEW

**Comment by rustyrussell** [comment]
Trivial rebase

**Comment by rustyrussell** [comment]
Rebase, crash fix for offers (which don't have payment_metadata yet!).

**Comment by rustyrussell** [comment]
OK, this time I added a much bigger expiry to the canned invoice!!

**Comment by rustyrussell** [comment]
Trivial rebase, to fix simple conflict and get master flake fixes.

**Comment by rustyrussell** [comment]
Trivial rebase, and a real fix for subd memleak report!

Ack https://github.com/ElementsProject/lightning/commit/5f89486855e00c84b28c626e65cf0f748cdfc8a5

**Comment by rustyrussell** [comment]
Rebase again: we didn't set the new field to NULL in the newly re-introduced legacy onion path.

Ack https://github.com/ElementsProject/lightning/commit/9f4dfc8dc746f2f245738e5a329b6ecccf57d897

File: common/test/run-bolt11.c [hunk 1]
```diff
@@ -118,13 +118,16 @@ static void test_b11(const char *b11str,
 	if (b11->min_final_cltv_expiry == 18)
 		return;
 
+	/* Also blockstream store example signature doesn't match? */
 	/* Re-encode to check */
 	reproduce = bolt11_encode(tmpctx, b11, false, test_sign, NULL);
+#if 0
```
Comment (cdecker) [inline common/test/run-bolt11.c]: Missing a `FIXME`?

**Review by cdecker — APPROVED** [review]
ACK 91932be70543707d6e51909c1aac9cb5669f9806

File: common/test/run-bolt11.c (hunk 1 above)
Comment (cdecker) [reply common/test/run-bolt11.c]: Removed again in the second to last commit in this PR.

File: lightningd/closing_control.c [hunk 2]
```diff
@@ -684,11 +685,11 @@ static struct command_result *json_close(struct command *cmd,
 				       channel->peer->their_features,
 				       OPT_SHUTDOWN_ANYSEGWIT);
 	if (!valid_shutdown_scriptpubkey(channel->shutdown_scriptpubkey[LOCAL],
-					 anysegwit)) {
+					 anysegwit, !deprecated_apis)) {
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply lightningd/closing_control.c:688]: Future work! This flag will be always set when we implement it, which may Just Happen if we're slow enough:)

**Pull request merged**